utils/update_checkout/update_checkout/update_checkout.py

```diff
@@ -60,18 +60,33 @@ def check_parallel_results(results, op):
     parallel implementation.
     """
 
-    fail_count = 0
     if results is None:
         return 0
+
+    # Remove non-errors
+    results = [r for r in results if r]
+
+    if results:
+        print(f"======{op} FAILURES======", file=sys.stderr)
+
+    messages = []
     for r in results:
-        if r is not None:
-            if fail_count == 0:
-                print("======%s FAILURES======" % op)
-            print("%s failed (ret=%d): %s" % (r.repo_path, r.ret, r))
-            fail_count += 1
-            if r.stderr:
-                print(r.stderr)
-    return fail_count
+        try:
+            raise r
+        except KeyError as e:
+            messages.append(f"Failed to look up {e} in scheme\n")
+        except Exception:
```

Comment:
I just realized you can make this a bit simpler and avoid the "`raise` for pattern matching" which is a bit quirky
```suggestion
        if isinstance(r, KeyError):
            messages.append(f"Failed to look up {r} in scheme\n")
        else:
```

Reply:
I wasn't sure which read better. It should probably be single `JobError` with a `__str__` method that just prints the right thing so this can just iterate over the list and print it without having to figure out what happened in the job that failed (even the KeyError here could lead to a misleading error message), but that is a bigger PR.